src/lib.rs

```diff
@@ -190,21 +192,15 @@ impl DocData {
             // example:: -> example
             name: root_def.qualname[..(name_len - 2)].to_string(),
             docs: root_def.docs.clone(),
-            modules: Vec::new(),
+            metadata: Vec::new(),
         };
 
-        let data = DocData::build_data(config, root_id, &mut krate)?;
+        DocData::build_data(config, root_id, &mut krate)?;
 
-        Ok(DocData { krate, data })
+        Ok(DocData { krate })
     }
 
-    fn build_data(
-        config: &Config,
-        root_id: analysis::Id,
-        krate: &mut Crate,
-    ) -> Result<HashMap<String, Item>> {
-        let mut data = HashMap::new();
-
+    fn build_data(config: &Config, root_id: analysis::Id, krate: &mut Crate) -> Result<()> {
```

Comment:
So, this still ends up leaving this function and taking a `&mut Crate`; what do you think about possibly inlining this into the constructor? does it being its own function make sense still?